apps/omg/lib/omg/state/core.ex

```diff
@@ -403,18 +403,29 @@ defmodule OMG.State.Core do
   end
 
   def exit_utxos([%{tx_hash: _} | _] = piggybacks, %Core{utxos: utxos} = state) do
-    piggybacks
-    |> Enum.map(fn %{tx_hash: tx_hash, output_index: oindex} ->
-      # oindex in contract is 0-7 where 4-7 are outputs
-      oindex = oindex - 4
-
-      utxos
-      |> Map.to_list()
-      |> Enum.find(&match?({Utxo.position(_, _, ^oindex), %Utxo{creating_txhash: ^tx_hash}}, &1))
-    end)
-    |> Enum.filter(&(&1 != nil))
-    |> Enum.map(fn {position, _} -> position end)
-    |> exit_utxos(state)
+    {piggybacks_of_unknown_utxos, piggybacks_of_known_utxos} =
+      piggybacks
+      |> Enum.map(fn %{tx_hash: tx_hash, output_index: oindex} = piggyback ->
```

Comment:
maybe these big pipes are not the optimal approach to this? what if you pull the implementation out into a private function and document the function separately + specs?

Reply:
ah, I concur about pulling the implementation out to a `defp`, but I'd leave all 3 pipes in the diff in :)

Reply:
done